### PR TITLE
add first test to check pages slide

### PR DESCRIPTION
**Pull request by Alexandru-Bulai**
# PR 13-03-24 : First test to check pages slide
## Overview
In this PR we implement a function that is going to keep track of the pages in the highlight section 
## Key Changes :
 1) indexDom.js : 
    - Added necessary tests to check core logic of the function 
    - Add a function that is going to keep lets the user know in which of the highlight section he's 
2) vite.config.js : 
    - Added happy-dom as environment to test the DOM

# Additional notes
For more info about happy-dom. (https://github.com/capricorn86/happy-dom)

### CODE REVIEW

**Comment by vladiiancu** [comment]
Choosing Happy Dom over JSDOM - what is the reason? To play with another tool (variety) or something was not working with JSDOM (better fit)?

File: src/test/indexDom.test.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  test('Check decreaseing count', () => {
+    moveMenu(false, true)
+    expect(pageCount).toBe(0)
+  })
+
+  test('Ensure nothing else can affect count', () => {
+    moveMenu(false, false)
+    expect(pageCount).toBe(0)
+  })
```
Comment (vladiiancu) [inline src/test/indexDom.test.js:41]: It seems that there is one case missing: true + true

**Review by vladiiancu — COMMENTED** [review]
As far as I can tell, you have adopted TDD here. You defined a function in the test spec, wrote test cases and run them all to pass. 
And your next step will be to move the moveMenu function outside, right? 

Very interesting

File: src/test/indexDom.test.js (hunk 1 above)
Comment (Alexandru-Bulai) [reply src/test/indexDom.test.js:38]: Yes, moving the function outside îs going to be the next step :)

**Comment by Alexandru-Bulai** [comment]
> Choosing Happy Dom over JSDOM - what is the reason? To play with another tool (variety) or something was not working with JSDOM (better fit)?

I wanted to try another tool for this project, but JSDOM works well too

File: src/test/indexDom.test.js (hunk 1 above)
Comment (vladiiancu) [reply src/test/indexDom.test.js:41]: @Alexandru-Bulai have you made any changes regarding this?

Comment (Alexandru-Bulai) [reply src/test/indexDom.test.js:41]: @vladiiancu  Yessir, i added the test on the last Commit if im not mistaking

**Review by vladiiancu — APPROVED** [review]

**Pull request merged**